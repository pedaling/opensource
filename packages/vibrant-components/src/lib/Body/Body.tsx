import { Text } from '../Text';
import { withBodyVariation } from './BodyProps';

export const Body = withBodyVariation(({ innerRef, children, ...textProps }) => (
  <Text ref={innerRef} {...textProps}>
    {children}
  </Text>
));
