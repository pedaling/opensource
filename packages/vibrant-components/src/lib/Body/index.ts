export { Body } from './Body';
export type { BodyProps } from './BodyProps';
