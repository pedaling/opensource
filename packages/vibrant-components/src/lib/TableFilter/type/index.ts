export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
