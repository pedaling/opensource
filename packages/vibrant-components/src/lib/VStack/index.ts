export { VStack } from './VStack';
export type { VStackProps } from './VStackProps';
