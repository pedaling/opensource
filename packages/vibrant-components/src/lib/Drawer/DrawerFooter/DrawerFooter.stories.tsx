import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { DrawerFooter } from './DrawerFooter';

export default {
  title: 'DrawerFooter',
  component: DrawerFooter,
  args: {},
} as ComponentMeta<typeof DrawerFooter>;

export const Basic: ComponentStory<typeof DrawerFooter> = props => <DrawerFooter {...props} />;
