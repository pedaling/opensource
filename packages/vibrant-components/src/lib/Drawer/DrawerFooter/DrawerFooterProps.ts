import type { ReactElement } from 'react';
import { withVariation } from '@vibrant-ui/core';

export type DrawerFooterProps = {
  children: ReactElement | ReactElement[];
  testId?: string;
};

export const withDrawerFooterVariation = withVariation<DrawerFooterProps>('DrawerFooter')();
