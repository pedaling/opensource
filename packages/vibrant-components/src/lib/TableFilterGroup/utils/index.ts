import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
