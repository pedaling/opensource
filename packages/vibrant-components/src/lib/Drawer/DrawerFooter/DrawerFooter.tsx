import { Box } from '@vibrant-ui/core';
import { Divider } from '../../Divider';
import { withDrawerFooterVariation } from './DrawerFooterProps';

export const DrawerFooter = withDrawerFooterVariation(({ testId = 'drawer-footer', children }) => (
  <>
    <Divider direction="horizontal" kind="default" />
    <Box data-testid={testId} py={12} px={20}>
      {children}
    </Box>
  </>
));
