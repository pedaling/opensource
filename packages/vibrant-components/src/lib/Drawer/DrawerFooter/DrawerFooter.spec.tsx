import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Body } from '../../Body';
import { VStack } from '../../VStack';
import { DrawerFooter } from './DrawerFooter';

describe('<DrawerFooter />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when DrawerFooter is rendering', () => {
    beforeEach(async () => {
      renderer = render(
        <DrawerFooter>
          <VStack>
            <Body level={3}>Drawer Footer test</Body>
          </VStack>
        </DrawerFooter>
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
