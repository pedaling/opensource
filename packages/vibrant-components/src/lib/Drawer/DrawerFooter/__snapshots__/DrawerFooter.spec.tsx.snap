// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerFooter /> when DrawerFooter is rendering match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 20px;
  padding-right: 20px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.8125rem;
  line-height: 1.125rem;
  font-weight: 400;
}

<div>
  <div
    class="emotion-0"
    kind="default"
  />
  <div
    class="emotion-1"
    data-testid="drawer-footer"
  >
    <div
      class="emotion-2"
    >
      <span
        class="emotion-3"
        data-testid="body"
      >
        Drawer Footer test
      </span>
    </div>
  </div>
</div>
`;
